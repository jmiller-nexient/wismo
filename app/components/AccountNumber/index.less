@import '../../styles/themes/index.less';

.dte-wismo-account-number {
  color: @dte-gray-500;
  font-family: @dte-wismo-font-family;
  font-size: @dte-wismo-font-size-1;
}
