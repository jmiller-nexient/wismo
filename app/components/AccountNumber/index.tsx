import * as React from 'react';

import formatAccountNumber from './formatAccountNumber';

interface IAccountNumberProps {
  value: string;
}

const AccountNumber: React.FC<IAccountNumberProps> = (props: IAccountNumberProps): React.ReactElement => {
  const { value } = props;

  if (value.length !== 12) {
    throw new Error('account numbers must be 12 numeric digits.');
  }

  const formattedAccountNumber = formatAccountNumber(value);

  return <span>{formattedAccountNumber}</span>;
};

export default AccountNumber;
