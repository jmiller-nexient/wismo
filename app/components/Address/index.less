@import '../../styles/themes/index.less';

@address-prefix-cls: ~'@{dte-prefix}-address';

.@{address-prefix-cls} {
  line-height: 1.375rem;
}
