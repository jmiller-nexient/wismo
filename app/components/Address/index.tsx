import React from 'react';

import './index.less';

interface IAddressProps {
  readonly city: string;
  readonly line1: string;
  readonly line2?: string | null;
  readonly state: string;
  readonly zip: string;
}

const Address: React.FC<IAddressProps> = (props: IAddressProps): React.ReactElement => {
  const { city, line1, line2, state, zip } = props;

  return (
    <address className="dte-wismo-address">
      {line1 + (line2 ? `, ${line2}` : '')}
      <br />
      {`${city}, ${state} ${zip}`}
    </address>
  );
};

export default Address;
