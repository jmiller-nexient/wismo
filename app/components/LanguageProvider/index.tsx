import React, { useState } from 'react';
import { IntlProvider } from 'react-intl';

// Import i18n messages
import { translationMessages } from '../../i18n';

import { DEFAULT_LOCALE, SUPPORTED_LOCALES } from '../../locales';

interface ILanguageProviderProps {
  children?: React.ReactNode;
  locale?: string;
}

type SupportedLocalesType = typeof SUPPORTED_LOCALES;

interface ILanguageState {
  changeLanguage: (language: string) => void;
  language: string;
  supportedLanguages: SupportedLocalesType;
}

const LanguageContext = React.createContext<ILanguageState | null>(null);

const LanguageConsumer = LanguageContext.Consumer;

const LanguageProvider: React.FC<ILanguageProviderProps> = (props: ILanguageProviderProps): React.ReactElement => {
  const { children, locale = DEFAULT_LOCALE } = props;

  const [language, setLanguage] = useState<string>(locale);

  // TODO: Remove the slice operation if regional variants are being supported
  //       E.g. en-CA, en-GB, en-US instead of just en
  const messages = translationMessages[language.length > 2 ? language.slice(0, 2) : language];

  return (
    <LanguageContext.Provider
      value={{
        changeLanguage: setLanguage,
        language,
        supportedLanguages: SUPPORTED_LOCALES,
      }}
    >
      <IntlProvider key={locale} locale={locale} messages={messages}>
        {React.Children.only(children)}
      </IntlProvider>
    </LanguageContext.Provider>
  );
};

export { LanguageConsumer, LanguageContext, LanguageProvider };
