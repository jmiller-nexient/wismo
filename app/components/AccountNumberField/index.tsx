import React from 'react';

import BaseFormField from 'components/BaseField';
import AccountNumber from '../AccountNumber';

import messages from './messages';

interface IAccountNumberFieldProps {
  accountNumber: string;
}

const AccountNumberField: React.FC<IAccountNumberFieldProps> = (props: IAccountNumberFieldProps): React.ReactElement => {
  const { accountNumber } = props;

  return (
    <BaseFormField label={messages.accountNumberFieldLabel}>
      <AccountNumber value={accountNumber}/>
    </BaseFormField>
  );
};

export default AccountNumberField;
