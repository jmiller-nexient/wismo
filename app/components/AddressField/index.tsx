import React from 'react';

import Address from 'components/Address';
import BaseFormField from 'components/BaseField';

import IAddress from 'types/address';

import messages from './messages';

interface IAddressFieldProps {
  address: IAddress;
}

const AddressField: React.FC<IAddressFieldProps> = (props: IAddressFieldProps): React.ReactElement => {
  const { address } = props;

  return (
    <BaseFormField label={messages.addressFieldLabel}>
      <Address {...address}/>
    </BaseFormField>
  );
};

export default AddressField;
