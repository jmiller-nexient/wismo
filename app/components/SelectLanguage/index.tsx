import React, { ChangeEvent, useContext } from 'react';

import { LanguageContext } from 'components/LanguageProvider';

const SelectLanguage: React.FC = (): React.ReactElement => {
  const context = useContext(LanguageContext);

  if (!context) {
    throw new Error('useContext can only be used inside LanguageProvider');
  }

  const { changeLanguage, language, supportedLanguages } = context;

  const renderOptions = (): React.ReactElement[] =>
    Object.keys(supportedLanguages).map(
      (key: string): React.ReactElement => (
        <option key={key} value={key}>
          {supportedLanguages[key]}
        </option>
      ),
    );

  const selectionChanged = (event: ChangeEvent<HTMLSelectElement>): void => changeLanguage(event.target.value);

  return (
    <select onChange={selectionChanged} value={language}>
      {renderOptions()}
    </select>
  );
};

export default SelectLanguage;
