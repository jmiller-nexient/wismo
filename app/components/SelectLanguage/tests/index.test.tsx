import React, { ReactElement, useContext } from 'react';
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';

import { LanguageProvider, LanguageContext } from 'components/LanguageProvider';

import SelectLanguage from '..';

const FakeConsumer: React.FC = (): React.ReactElement => {
  const context = useContext(LanguageContext);

  return (
    <div id="current-language">
      {context?.language}
    </div>
  );
};

describe('<SelectLanguage />', () => {
  it('should change global state when a new value is selected', () => {
    const component = mount(
      <LanguageProvider locale="en">
        <div>
          <SelectLanguage />
          <FakeConsumer />
        </div>
      </LanguageProvider>
    );

    const select = component.find('select');

    select.simulate('change', {target: { value: 'es'}});

    const consumer = component.find('div#current-language').at(0);

    expect(consumer.text()).toEqual('es');
  });

  it('should render correctly when wrapped in LanguageProvider', () => {
    const tree: ReactElement = renderer.create(
      <LanguageProvider>
        <SelectLanguage />
      </LanguageProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should throw an error when not wrapped in LanguageProvider', () => {
    expect(() => shallow(<SelectLanguage />)).toThrowError();
  });
});
