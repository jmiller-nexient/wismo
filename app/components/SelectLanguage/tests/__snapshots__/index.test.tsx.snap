// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectLanguage /> should render correctly when wrapped in LanguageProvider 1`] = `
<select
  onChange={[Function]}
  value="en"
>
  <option
    value="en"
  >
    English
  </option>
  <option
    value="es"
  >
    Spanish
  </option>
</select>
`;
